test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/MetadataBindingTests.cs

```diff
@@ -0,0 +1,215 @@
+        }
+
+        #region Tests
+        [Fact]
+        public void MetadataBinderProcessSkipWithNullReturnsNull()
+        {
+            long? result = MetadataBinder.ProcessSkip(null);
+            result.Should().Be(null);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessSkipWithNegativeNumberShouldThrow()
+        {
+            Action metadataBinderAction = () => MetadataBinder.ProcessSkip(-1);
+            metadataBinderAction.ShouldThrow<ODataException>().WithMessage(
+                ODataErrorStrings.MetadataBinder_SkipRequiresNonNegativeInteger("-1"));
+        }
+
+        [Fact]
+        public void MetadataBinderProcessSkipWithPositiveNumberShouldReturnNumber()
+        {
+            long? result = MetadataBinder.ProcessSkip(1);
+            result.Should().Be(1);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessTopWithNullReturnsNull()
+        {
+            long? result = MetadataBinder.ProcessTop(null);
+            result.Should().Be(null);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessTopWithNegativeNumberShouldThrow()
+        {
+            Action metadataBinderAction = () => MetadataBinder.ProcessTop(-1);
+            metadataBinderAction.ShouldThrow<ODataException>().WithMessage(
+                ODataErrorStrings.MetadataBinder_TopRequiresNonNegativeInteger("-1"));
+        }
+
+        [Fact]
+        public void MetadataBinderProcessTopWithPositiveNumberShouldReturnNumber()
+        {
+            long? result = MetadataBinder.ProcessTop(1);
+            result.Should().Be(1);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessQueryOptionsWithNullBindStateShouldThrow()
+        {
+            BindInfo bindInfo = new BindInfo();
+
+            // Test null bind state
+            Action metadataBinderAction = () => MetadataBinder.ProcessQueryOptions(null, bindInfo.BindMethod);
+            metadataBinderAction.ShouldThrow<ODataException>(
+                ODataErrorStrings.MetadataBinder_QueryOptionsBindStateCannotBeNull);
+
+            // Test bind state that doesn't have query options populated
+            metadataBinderAction = () => MetadataBinder.ProcessQueryOptions(bindInfo.BindingState, bindInfo.BindMethod);
+            metadataBinderAction.ShouldThrow<ODataException>(
+                ODataErrorStrings.MetadataBinder_QueryOptionsBindStateCannotBeNull);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessQueryOptionsWithNullBindMethodShouldThrow()
+        {
+            BindInfo bindInfo = new BindInfo();
+
+            Action metadataBinderAction = () => MetadataBinder.ProcessQueryOptions(bindInfo.BindingState, null);
+            metadataBinderAction.ShouldThrow<ODataException>(
+                ODataErrorStrings.MetadataBinder_QueryOptionsBindMethodCannotBeNull);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessQueryOptionsWithBindMethodReturnNullShouldReturnEmptyList()

```

Comment:
It's the name of a delegate so there's not actually double return validation. I've updated the name of the test for better clarity.